addon/components/materialize-tabs.js

```diff
@@ -0,0 +1,87 @@
+import Ember from 'ember';
+import layout from '../templates/components/materialize-tabs';
+
+export default Ember.Component.extend({
+  layout: layout,
+  tagName: 'ul',
+  tabWidth: 2,
+  classNames: ['tabs'],
+
+  tabComponents: [],
+  activeTab: null,
+
+  registerTab: function (tab) {
+    var tabs = this.get('tabComponents');
+    tabs.addObject(tab);
+  },
+
+  setIndicatorPosition: function (left, right, animateUpdate=true) {
+    var $i1 = this.$('.indicator_1');
+    var $i2 = this.$('.indicator_2');
+
+    var destOptions = {
+      left: `${left}px`,
+      right: `${right}px`
+    };
+
+    var animationConfig = {
+        duration: 200,
+        easing: "swing"
+    };
+    if (!animateUpdate) {
+      this.$('.indicator').css(destOptions);
+    }
+    else {
+      $i1.velocity(destOptions, animationConfig);
+      $i2.velocity(destOptions, Ember.$.extend({delay: animationConfig.duration/4}, animationConfig));
+    }
+  },
+
+  updateIndicators: function (animateUpdate=true) {
+    var activeTab = this.get('activeTab');
+    var tabSetRect = this.element.getBoundingClientRect();
+    var activeTabRect = activeTab.element.getBoundingClientRect();
+    this.setIndicatorPosition(activeTabRect.left - tabSetRect.left, tabSetRect.right - activeTabRect.right, animateUpdate);
+  },
+
+  setActiveTab: function (tab, animateUpdate=true) {
+    var currentActiveTab = this.get('activeTab');
+    if (!currentActiveTab || (currentActiveTab.element.id !== tab.element.id)) {
+      this.set('activeTab', tab);
+      tab.notifyPropertyChange('isActive');
+      if (currentActiveTab) {
+        currentActiveTab.notifyPropertyChange('isActive');
+      }
+      this.updateIndicators(animateUpdate);
+    }
+  },
+
+  didInsertElement: function () {
+    this._super(...arguments);
+    var router = this.container.lookup('router:main');
+    if (router) {
+      var handlers = router.router.state.handlerInfos;
+      var routeName = handlers[handlers.length - 1].name;
+      var active = this.get('tabComponents').findBy('dest', routeName);
+      this.setActiveTab(active, false);
+    }
+
+    this.onWindowResize = function () {
+      Ember.run.debounce(this, this.updateIndicators, false, 50);
+    }.bind(this);
```

Comment:
If the window is resized, the "active" indicator needs to be re-positioned.

**Note to self:** we probably would benefit from a `window.resize` service of some sort, so that a single DOM event listener can be shared across many components